nerfstudio/pipelines/base_pipeline.py

```diff
@@ -108,9 +108,14 @@ def device(self):
         return self.model.device
 
     def load_state_dict(self, state_dict: Mapping[str, Any], strict: bool = True):
-        model_state = {key[len("_model.") :]: value for key, value in state_dict.items() if key.startswith("_model.")}
+        # remove the "_model." or "_model.module." prefix from key
+        model_state = {
+            key[len("_model.module.") :] if key.startswith("_model.module.") else key[len("_model.") :]: value
+            for key, value in state_dict.items()
+            if key.startswith("_model.")
+        }
```

Comment:
This comprehension is getting kinda messy and hard to read. Can you rewrite as a for loop